src/targets/gpu/target.cpp

```diff
@@ -50,8 +56,9 @@ std::vector<pass> target::get_passes(migraphx::context& gctx) const
         dead_code_elimination{},
         fuse_ops{&ctx},
         dead_code_elimination{},
-        write_literals{&ctx},
-        memory_coloring{"hip::allocate"},
+        write_literals{&ctx}, 
+        pre_scheduling{weight_func, num_of_streams, insert_instruction_gpu{}},
+        memory_coloring{"hip::allocate", num_of_streams, find_concur_gpu{}},
```

Comment:
`num_of_streams` should be passed to `find_concur_gpu` constructor and not `memory_coloring`. If you dont want to change it, that is fine. I am just leaving this as a note for changes that need to be made.

Reply:
I choose no chance for now.